pytube/contrib/channel.py

```diff
@@ -183,19 +249,119 @@ def _extract_videos(raw_json: str) -> Tuple[List[str], Optional[str]]:
             # if there is an error, no continuation is available
             continuation = None
 
+        # only extract the video ids from the video data
+        videos_url = []
+        try:
+            # Extract id from videos and live
+            for x in videos:
+                videos_url.append(f"/watch?v="
+                                  f"{x['richItemRenderer']['content']['videoRenderer']['videoId']}")
+        except (KeyError, IndexError, TypeError):
+            # Extract id from short videos
+            for x in videos:
```

Comment:
what would happen if we get another exception within this loop?

Reply:
So far Youtube is not returning any value that could generate exceptions. But how about adding an "**ExtractError**" to make it easier to track the error in a future update?